typing/typecore.ml

```diff
@@ -4536,8 +4536,9 @@ and type_constraint_expect
     nodes for the newtype properly linked.
 *)
 and type_newtype
-  : type a. _ -> _ -> _ -> (Env.t -> a * type_expr) -> a * type_expr =
-  fun loc env name type_body ->
+  : type a. _ -> _ -> (Env.t -> a * type_expr) -> a * type_expr =
+  fun env name type_body ->
+  let { txt = name; loc = name_loc } : _ Location.loc = name in
```

Comment:
Remarks (that can be safely discarded): 
* this pattern could be moved to `fun env { txt = name; loc _ make_loc} type_body ->`.
* if needed the type annotation could be moved to the function type.

Reply:
I have adopted the suggestion.